src/main.ts

```diff
@@ -50,6 +52,9 @@ async function run(): Promise<void> {
 }
 
 async function post(): Promise<void> {
+  if (state.actionStartedAt) {
```

Comment:
If I understand this correctly, I don't think you actually need this. The comments are posted by the sonarqube server not by the agent so you can remove all sonar comments and then after that the server will post the comments. I don't see any harm with this though.